scripts/runner_test.go

```diff
@@ -0,0 +1,45 @@
+package scripts
+
+import (
+	"os"
+	"strings"
+	"testing"
+)
+
+func TestLookPath(t *testing.T) {
+	tests := []struct {
+		file string
+		wantErr bool
+	}{
+		{"no-such-file", true},
+		{"fetch-uj-records.sh", false},
+		{"get-uid-map.sh", false},
+		{"splunk-to-segment.sh", false},
+		{"segment-mass-uploader.sh", false},
+	}
+	for _, tt := range tests {
```

Comment:
Is there a way to make the intention of the test more clear?
I'm reading this and I cannot tell for sure what we're testing.

Reply:
no sure how, is basically iterating over script names, passing them to LookPath and then checking that the returned string:

1. ends with the script name
2. points to an existent, executable, regular file.

The `if`s are quire straight forward, so I'm not sure how to improve that. Permaps I should rename the `file` variable to indicate better its a script name.

Reply:
I think those loop and ifs go against the principles you're advocating for about writing simple test cases.

I'm now thinking that what makes the tests less obvious to comprehend is that we're not asserting for what we're after, but instead checking the absence of the complimenting condition. This also makes the tests less usable as a tool for understanding the code itself.

Reply:
I think most of the things you mention are idioms that arise from how the Go `testing` module is built.

The loop is equivalent to the `pytest.parameterize` decorator, but there are no decorators in Go, and in general, more of a tendency to do things explicitly rather then implicitly via e.g decorators. 

I think this loop is pretty much a standard idiom, given that it is essentially auto-generated for me by VScode. We should view the function more like a complete test suite running individual sub tests via `t.Run` rather then as a single test case.

As for the assertions - I guess I could come up with abstractions to make them look more like the assertions you are used to, but ultimately I think we are better off learning how to read code written to the existing standard library API.

I'm willing to try concrete suggestions, if you have any.

Reply:
https://pkg.go.dev/github.com/stretchr/testify/assert

I see this package for assertions which I think might make things look more straightforward. 
I don't think we should hold merging this one at this point, but I think this is something that could make our tests and code more maintainable.

In such case, we can replace this:
```
if !tt.wantErr {
	t.Errorf("LookPath() error = %v, wantErr %v", err, tt.wantErr)
}
```
With this:
```
assert.Truef(tt.wantErr, "LookPath() error = %v, wantErr %v", err, tt.wantErr)
```

There are assertions there for errors that can probably be used in this case which are more appropriate than asserting for a boolean value, but it's already more straightforward as far as I see it.

Reply:
@yftacherzog please have a 2nd look, using testify now

Reply:
What's the rationale behind using a combination of assert and require?

Reply:
assert logs the error and keeps going while require exists the test immediately.